jasper_reports/JasperReports/browse_data_generator.py

```diff
@@ -368,6 +368,8 @@ def generateCsvRecord(self, record, records, row, path, fields, sequence,
                           subsequence, copy):
         # One field (many2one, many2many or one2many) can appear several times
         # Process each "root" field only once by using a set.
+        if not record:
+            return
```

Comment:
Can you give a use case where the problem causes error?

Reply:
It occurs when a report has a table who is fetched with a one2many field, and you print with a record with the one2many empty. The error is because the csv created in /tmp puts a False value in id column

> XX-XX,XX-XX,XX-XX,XX-XX,XX-XX,**id**,XX-XX,XX-XX,XX-XX,
> ,,,,,**False**,,,,

If this solution can cause problems another solution is change:
>            if field == 'id':
>                # Check for field 'id' because we can't find it's
>                # type in _columns
>                value = str(value)
>            elif value in (False, None):
>                value = ''
to
>            if value in (False, None):
>                value = ''
>            elif field == 'id':
>                # Check for field 'id' because we can't find it's
>                # type in _columns
>                value = str(value)

P.S.: this is the raised error
>UserError: (u"Report Error\nFailed to invoke method execute in class com.nantic.jasperreports.JasperServer: net.sf.jasperreports.engine.JRException: Unable to get value for field 'id' of class 'java.lang.Integer': ", '')